nixos/modules/virtualisation/libvirtd.nix

```diff
@@ -335,6 +397,15 @@ in
           ln -s --force ${ovmfpackage}/FV/AAVMF_VARS.fd /run/${dirName}/nix-ovmf/
           ln -s --force ${ovmfpackage}/FV/OVMF_VARS.fd /run/${dirName}/nix-ovmf/
         '')}
+
+        # Symlink hooks to /var/lib/libvirt
+        ${concatStringsSep "\n" (map (driver:
+          ''
+          mkdir -p /var/lib/${dirName}/hooks/${driver}.d
+          rm -rf /var/lib/${dirName}/hooks/${driver}.d/*
```

Comment:
Why do we symlink the hooks together in nix and then symlink that directory to hooks/driver.d? Would that be cleaner?

Reply:
One disadvantage of your proposal is that hooks managed outside NixOS is impossible.

What do you mean by clean?

Reply:
Can we place symlink in the nix-store? If so that might be an approach.

Managing hooks outside NixOS is currently impossible, i see two approachs to "fix" this:

1.  Gate nixos managed hooks behind an enable option, i.e give the choice between nixos managed hooks or manually managed hooks.
2. Only delete hooks that are direct symlink to the nix-store while regenerating, this allow for manual hook to be implemented, even linking to the nix store via an indirect symlink. (Which you want to do if you are doing this because you need a gcroot.)

My preference would be option 2 i guess.

Reply:
> One disadvantage of your proposal is that hooks managed outside NixOS is impossible.

We can always add an option to define custom hooks which would be the nixos way.

Reply:
Then clearing the directory should be no problem, since the hooks are recreated.

Reply:
Marking this as resolved :)